### PR TITLE
cancel the challenge if OAuth user credential creation is cancelled

### PR DESCRIPTION
**Pull request by njarecha**
Cancelling the challenge will give user a new `ArcGISChallengeCancellationError` with underlying error.

### CODE REVIEW

File: Sources/ArcGISToolkit/Components/Authentication/Authenticator.swift [hunk 1]
```diff
@@ -50,7 +50,11 @@ extension Authenticator: ArcGISAuthenticationChallengeHandler {
         
         // Create the correct challenge type.
         if let configuration = oAuthUserConfigurations.first(where: { $0.canBeUsed(for: challenge.requestURL) }) {
-            return .continueWithCredential(try await OAuthUserCredential.credential(for: configuration))
+            do {
+                return .continueWithCredential(try await OAuthUserCredential.credential(for: configuration))
+            } catch is CancellationError {
+                return .cancel
+            }
```
Comment (rolson) [inline Sources/ArcGISToolkit/Components/Authentication/Authenticator.swift:57]: I don't understand why this change is needed

Comment (njarecha) [reply Sources/ArcGISToolkit/Components/Authentication/Authenticator.swift:57]: So the request will fail with `ArcGISChallengeCancellationError with underlying error` instead of `CancellationError`.

Comment (rolson) [reply Sources/ArcGISToolkit/Components/Authentication/Authenticator.swift:57]: should It be catching `ArcGISChallengeCancellationError` then?

Comment (rolson) [reply Sources/ArcGISToolkit/Components/Authentication/Authenticator.swift:57]: Never mind, I think I understand. I do think a comment would be good here probably

Comment (njarecha) [reply Sources/ArcGISToolkit/Components/Authentication/Authenticator.swift]: done

**Review by rolson — APPROVED** [review]

**Review by dfeinzimer — APPROVED** [review]

**Comment by njarecha** [comment]
Thanks @rolson and @dfeinzimer!
